qr-code/src/main/java/com/adyen/checkout/qrcode/QRCodeComponent.kt

```diff
@@ -98,7 +110,7 @@ class QRCodeComponent(application: Application, configuration: QRCodeConfigurati
         // Immediately request a new status if the user resumes the app
         lifecycleOwner.lifecycle.addObserver(object : BaseLifecycleObserver() {
             override fun onResume() {
-                statusRepository.updateStatus()
+                if (isPollingEnabled) statusRepository.updateStatus()
```

Comment:
This seems like something that could have a better separation. Maybe the `StatusRepository` should only handle it if it's currently polling?

Reply:
Agreed, I'll add a status flag inside `StatusRepository`